core/internal/filetransfer/file_transfer_retry_policy.go

```diff
@@ -13,13 +14,28 @@ func FileTransferRetryPolicy(
 	resp *http.Response,
 	err error,
 ) (bool, error) {
-	// Abort on any error from the HTTP transport.
-	//
-	// This happens if reading the file fails, for example if it is a directory
-	// or if it doesn't exist. retryablehttp's default policy for this situation
-	// is to retry, which we do not want.
 	if err != nil {
-		return false, err
+		switch {
+		// Retry dial tcp <IP>: i/o timeout errors.
+		//
+		// Cloud storage providers sometimes return such i/o timeout errors when rate limiting
+		// without specifying the error type to prevent potential malicious attacks.
+		// See https://wandb.atlassian.net/browse/WB-18702 for more context.
```

Comment:
I don't see "dial tcp" in the linked issue. Is it really returned by a cloud storage provider, or is it a local error? Is there any official documentation about this behavior?